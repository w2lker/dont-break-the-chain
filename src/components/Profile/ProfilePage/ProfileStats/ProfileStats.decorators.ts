import withStyles from '@material-ui/core/styles/withStyles';
import { connect } from 'react-redux';
import { IProfileReducer } from '../../../../models/profile';

import ProfileStats, { IProfileStatsProps } from './ProfileStats';
import profileStatsStyles from './ProfileStats.styles';

const mapStateToProps = (state: any): IProfileStatsProps => {
  const profile: IProfileReducer = state.profile;
  return {
    total: profile.points,
    days: profile.longestChain,
  };
};

const mapDispatchToProps = null;

const connectedComponent = connect(mapStateToProps, mapDispatchToProps)(ProfileStats);

// @ts-ignore
const styledComponent = withStyles(profileStatsStyles)(connectedComponent);

export default styledComponent;
