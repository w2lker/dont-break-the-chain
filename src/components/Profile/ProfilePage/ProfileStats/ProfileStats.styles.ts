import { bon } from '../../../../utils/styleUtils';

const profileStatsStyles = {
  wrapper: {
    ...bon.setFlex('center', 'center'),
    margin: '20px 30px 0',
    padding: '40px 0 30px',
    boxShadow: '0px 4px 8px rgba(176, 190, 197, 0.24)',
  },
  statWrapper: {
    margin: '0 16px',
  },
  statValue: {
    ...bon.setFont(36, '#263238', 300, 'unset'),
    textAlign: 'left',
    margin: 0,
  },
  statSubtitle: {
    fontSize: 14,
  },
  statDescription: {
    ...bon.setFont(14, 'rgba(38, 50, 40, 0.72)', 400, '0.25px'),
    margin: 0,
    lineHeight: '20px',
  },
};

export default profileStatsStyles;
