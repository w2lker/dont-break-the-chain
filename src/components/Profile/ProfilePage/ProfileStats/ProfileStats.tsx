import React from 'react';
import defaultTexts from '../../../../contants/defaultTexts';

export interface IProfileStatsProps {
  total?: number;
  days?: number;
  classes?: any;
}

const ProfileStats: React.FC<IProfileStatsProps> = (props) => {
  const { total, days, classes } = props;
  if (!total && !days) {
    return null;
  }

  const texts = defaultTexts.profile.stats;

  return (
    <div className={classes.wrapper}>
      {total && (
        <div className={classes.statWrapper}>
          <h4 className={classes.statValue}>{total}</h4>
          <p className={classes.statDescription}>
            {texts.total.description}
          </p>
        </div>
      )}
      {days && (
        <div className={classes.statWrapper}>
          <h4 className={classes.statValue}>
            {days}&nbsp;
            <span className={classes.statSubtitle}>
              {texts.days.subtitle}
            </span>
          </h4>
          <p className={classes.statDescription}>
            {texts.days.description}
          </p>
        </div>
      )}
    </div>
  );
};

ProfileStats.defaultProps = {
  classes: {
    wrapper: '',
    statWrapper: '',
    statValue: '',
    statSubtitle: '',
    statDescription: '',
  },
};

export default ProfileStats;
