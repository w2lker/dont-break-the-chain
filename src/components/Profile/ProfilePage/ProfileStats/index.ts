import ProfileStats from './ProfileStats.decorators';

export default ProfileStats;
