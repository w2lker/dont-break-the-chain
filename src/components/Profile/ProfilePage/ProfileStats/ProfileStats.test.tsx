import React from 'react';

import { shallow } from 'enzyme';
import ProfileStats, { IProfileStatsProps } from './ProfileStats';
import profileStatsStyles from './ProfileStats.styles';

const classes = {
  wrapper: 'profile-stats-wrapper',
  statWrapper: 'profile-stats-stat-wrapper',
  statValue: 'profile-stats-stat-value',
  statSubtitle: 'profile-stats-stat-subtitle',
  statDescription: 'profile-stats-stat-description',
};

const classesKeys = Object.keys(classes);

describe('ProfileStats component', () => {

  const emptyProps: IProfileStatsProps = {
    classes,
  };

  const sampleProps: IProfileStatsProps = {
    classes,
    days: 33,
    total: 1453,
  };

  it('match snapshot', () => {
    const component = shallow(<ProfileStats {...sampleProps} />);
    expect(component.debug()).toMatchSnapshot();
  });

  it('skips render if no props passed', () => {
    const component = shallow(<ProfileStats {...emptyProps} />);
    expect(component).toEqual({});
  });

  describe('renders with okay props', () => {
    const component = shallow(<ProfileStats {...sampleProps} />);
    it('renders wrapper', () => {
      const wrapper = component.find(`.${classes.wrapper}`);
      expect(wrapper.length).toBe(1);
    });
    it('renders 2 scores', () => {
      const scoreWrappers = component.find(`.${classes.statWrapper}`);
      expect(scoreWrappers.length).toBe(2);
      const scoreValues = component.find(`.${classes.statWrapper} .${classes.statValue}`);
      expect(scoreValues.length).toBe(2);
    });
    it('renders total score', () => {
      const totalScore = component.find(`.${classes.statWrapper} .${classes.statValue}`).at(0);
      const totalScoreText = totalScore.text();
      // @ts-ignore
      expect(totalScoreText).toMatch(sampleProps.total.toString());
    });
    it('renders days amount', () => {
      const daysScore = component.find(`.${classes.statWrapper} .${classes.statValue}`).at(1);
      const daysScoreText = daysScore.text();
        // @ts-ignore
      expect(daysScoreText).toMatch(sampleProps.days.toString());
    });
  });

  it('renders with only total', () => {
    const props = {
      classes,
      total: 123,
    };
    const component = shallow(<ProfileStats {...props}/>);
    const stats = component.find(`.${classes.statWrapper}`);
    expect(stats.length).toBe(1);
    const statsValue = component.find(`.${classes.statValue}`);
    expect(stats.length).toBe(1);
    const statsText = statsValue.at(0).text();
    expect(statsText).toMatch(props.total.toString());
  });

  it('renders with only days', () => {
    const props = {
      classes,
      days: 13,
    };
    const component = shallow(<ProfileStats {...props}/>);
    const stats = component.find(`.${classes.statWrapper}`);
    expect(stats.length).toBe(1);
    const statsValue = component.find(`.${classes.statValue}`);
    expect(stats.length).toBe(1);
    const statsText = statsValue.at(0).text();
    expect(statsText).toMatch(props.days.toString());
  });
});

describe('ProfileStats styles', () => {
  it('contains principle classes', () => {
    expect(profileStatsStyles).toBeDefined();
    classesKeys.forEach((keyValue) => {
      // @ts-ignore
      expect(profileStatsStyles[keyValue]).toBeDefined();
    });
  });
});

describe('ProfileStats decorators', () => {
  test.todo('Test if all classes are passed to component');
  test.todo('Test if score and days are passed from connected store');
});
