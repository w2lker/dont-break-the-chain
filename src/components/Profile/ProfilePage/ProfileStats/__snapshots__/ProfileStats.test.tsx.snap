// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileStats component match snapshot 1`] = `
"<div className=\\"profile-stats-wrapper\\">
  <div className=\\"profile-stats-stat-wrapper\\">
    <h4 className=\\"profile-stats-stat-value\\">
      1453
    </h4>
    <p className=\\"profile-stats-stat-description\\">
      points received
    </p>
  </div>
  <div className=\\"profile-stats-stat-wrapper\\">
    <h4 className=\\"profile-stats-stat-value\\">
      33
      Â 
      <span className=\\"profile-stats-stat-subtitle\\">
        days
      </span>
    </h4>
    <p className=\\"profile-stats-stat-description\\">
      longest chain
    </p>
  </div>
</div>"
`;
